lib/chat_app/structs/room.ex

```diff
@@ -0,0 +1,37 @@
+defmodule ChatApp.Structs.Room do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @primary_key {:room_id, Ecto.UUID, autogenerate: true}
+  @foreign_key_type Ecto.UUID
+
+  schema "rooms" do
+    field(:room_name, :string)
+    field(:owner_name, :string)
+    field(:current_participants, {:array, :string})
+    field(:max_participants, :integer)
+    field(:is_private, :boolean)
+    field(:password, :string)
+    field(:expiry_timestamp, :utc_datetime)
+
+    has_many :messages, ChatApp.Structs.Message,
+    foreign_key: :room_id,
+    references: :room_id
```

Comment:
this doesn't seem to be properly formatted.

Reply:
I'll run a mix format for it and do another commit.